plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/stock-indicator/block.tsx

```diff
@@ -56,6 +58,42 @@ const getTextBasedOnStock = ( {
 	return __( 'Out of stock', 'woocommerce' );
 };
 
+/**
+ * Determines whether the stock indicator should be visible.
+ * Stock indicator is not visible when:
+ * - Product type is not allowed (variable, grouped, external)
+ * - Product is in stock, stock is not managed, and not on backorder
+ * - Product is a variation that is on backorder but backorder notification is disabled and stock is managed
+ *
+ * @param {ProductResponseItem} product Product object
+ * @return {boolean} True if stock indicator should be visible
+ */
+const isStockVisible = ( product: ProductResponseItem ): boolean => {
+	const productTypesWithoutStockIndicator = getSetting< string[] >(
+		'productTypesWithoutStockIndicator',
+		[ 'external', 'grouped', 'variable' ]
+	);
+
+	const isProductTypeAllowed = ! productTypesWithoutStockIndicator.includes(
+		product.type
+	);
+
+	const isInStockHidden =
+		product.is_in_stock &&
+		product.manage_stock === false &&
```

Comment:
Sorry, I might be getting confused, but why do we need `manage_stock`? I would expect that `is_in_stock`, `is_on_backorder` and `low_stock_remaining` variables would be enough to know whether we need to display the notice or not, no? :thinking:

Reply:
We need `manage_stock` in order to know if we should show the indicator when is in stock, because if the merchant checks the `Stock management` checkbox, the stock indicator will be visible even when `is_in_stock` is `true`. If that is not selected and `is_in_stock` is `true` we won't show the indicator.

![Screenshot 2024-12-05 at 4 16 56 PM](https://github.com/user-attachments/assets/61b71b21-8898-4549-a90a-7b1af75aaa5b)